### PR TITLE
Add: Generic calibration dataloader + forward function

### PR DESCRIPTION
**Pull request by rahul-tuli**
Adds: A generic calibratiion dataloader and a calibration forward function, for calibrating weights after training, (Post Training Quantization)
```python
import torch
from sparseml.pytorch.models import resnet50
from sparseml.pytorch.optim import QuantizationModifier


def random_dataloader(n=5):
    """
    A random generator
    """
    for _ in range(n):
        print(f"Trying to yield batch {_}")
        yield [torch.rand(32, 3, 224, 224)], {}


# Pretrained Model
model = resnet50(pretrained=True)

# Modifier init
modifier = QuantizationModifier(start_epoch=0)

# Modifier Application
modifier.apply(
    module=model,
    calibration_dataloader=random_dataloader(),
)

# Book-Keeping, the model should have non infinity weights
print(model(torch.rand(1, 3, 224, 224)))
print(model)
```
Can also pass in a Calibration function along with dataloader

```python
import torch
from sparseml.pytorch.models import resnet50
from sparseml.pytorch.optim import QuantizationModifier
from sparseml.pytorch.utils import  tensors_module_forward

def random_dataloader(n=5):
    """
    A random generator
    """
    for _ in range(n):
        print(f"Trying to yield batch {_}")
        yield [torch.rand(32, 3, 224, 224)], {}


# Pretrained Model
model = resnet50(pretrained=True)

# Modifier init
modifier = QuantizationModifier(start_epoch=0)

# Modifier Application
modifier.apply(
    module=model,
    calibration_dataloader=random_dataloader(),
    calibration_func=tensors_module_forward,
)

# Book-Keeping, the model should have non infinity weights
print(model(torch.rand(1, 3, 224, 224)))
print(model)
```

### CODE REVIEW

**Review by bfineran — CHANGES_REQUESTED** [review]
it looks like in your rebase you removed changes added in #548.

Additionally, what changes were added between this PR and #541

File: src/sparseml/pytorch/optim/modifier_quantization.py [hunk 1]
```diff
@@ -141,13 +158,15 @@ def __init__(
...
 
         self._modules_to_quantize = None
         self._qat_enabled = False
         self._quantization_observer_disabled = False
         self._bn_stats_frozen = False
-
+        self._calibration_dataloader = None
+        self._calibration_function = None
+        self._num_calibration_steps = num_calibration_steps
+        self._exclude_module_types = exclude_module_types
```
Comment (bfineran) [inline src/sparseml/pytorch/optim/modifier_quantization.py:169]: nit: move variables set from init params to group with line 160

**Comment by rahul-tuli** [comment]
> it looks like in your rebase you removed changes added in #548.
> 
> Additionally, what changes were added between this PR and #541

Great catch @bfineran Thanks.

File: src/sparseml/pytorch/optim/modifier_quantization.py (hunk 1 above)
Comment (rahul-tuli) [reply src/sparseml/pytorch/optim/modifier_quantization.py:169]: Addressed!

**Review by bfineran — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Pull request merged**